File: 14-longest-common-prefix/longest-common-prefix.java
class Solution {
    private String  helper(String s1,String s2){
        int n=Math.min(s1.length(),s2.length());
        StringBuilder sb=new StringBuilder();
        for(int i=0;i<n;i++){
            if(s1.charAt(i)==s2.charAt(i))sb.append(s1.charAt(i));
            else break;
        }
        return sb.toString();
    }
    public String longestCommonPrefix(String[] strs) {
         String ans=strs[0];
        
         for(int i=1;i<strs.length;i++){
            ans=helper(ans,strs[i]);
            
         }
         return ans;
    }
}

File: 14-longest-common-prefix/longest-common-prefix.cs
public class Solution {
 static String helper(String s,String t){
      int len=Math.Min(s.Length,t.Length);
      string ans="";
      for(int i=0;i<len;i++){
        if(s[i]==t[i])ans+=s[i];
        else break;
      }
      return ans;
  }
    public string LongestCommonPrefix(string[] strs) {
        string ans=strs[0];
        for(int i=1;i<strs.Length;i++){
            ans=helper(ans,strs[i]);
        }
        return ans;
    }
}